packages/backend-modules/publikator/lib/auphonic/index.ts

```diff
@@ -0,0 +1,249 @@
+import fetch, { Response } from 'node-fetch'
+import Debug from 'debug'
+import { ConnectionContext } from '@orbiting/backend-modules-types'
+
+const {
+  getAudioCover,
+} = require('@orbiting/backend-modules-documents/lib/meta')
+
+const { updateRepo } = require('../postgres')
+
+// @see https://auphonic.com/help/api/index.html
+const AUPHONIC_ENDPOINT = 'https://auphonic.com/api'
+
+const { AUPHONIC_BEARER, AUPHONIC_PRESET } = process.env
+
+const debug = Debug('publikator:lib:auphonic')
+
+type CallBody = {
+  preset?: string
+  metadata: {
+    title: string
+  }
+  output_basename: string
+  image?: string
+  preset_cover_image?: string
+}
+
+type Data = {
+  title: string
+  image?: string
+}
+
+type Document = {
+  content: {
+    meta: {
+      title: string
+      willBeReadAloud?: boolean
+    }
+  }
+}
+
+type AuphonicProduction = {
+  data: {
+    uuid: string
+  }
+}
+
+type AuphonicErrorPayload = {
+  repoId?: string
+  status?: number
+  url?: string
+}
+
+const handleResponse = async (res: Response): Promise<AuphonicProduction> => {
+  if (!res.ok) {
+    throw new AuphonicError(res.statusText, {
+      status: res.status,
+      url: res.url,
+    })
+  }
+
+  return res.json()
+}
+
+/**
+ * Wrapper to call Auphonic API.
+ */
+const call = async (
+  path: string,
+  method: 'POST' | 'GET' = 'POST',
+  body?: CallBody,
+) => {
+  const callDebug = debug.extend('call')
+  callDebug('path: %s', path)
+  callDebug('method: %s', method)
+  callDebug('body: %o', body)
+
+  const url = AUPHONIC_ENDPOINT + path
+  callDebug('url: %s', url)
+
+  const headers = {
+    ...(AUPHONIC_BEARER ? { Authorization: AUPHONIC_BEARER } : {}),
+    'Content-Type': 'application/json',
+  }
+  callDebug('headers: %o', headers)
+
+  const init = {
+    method,
+    headers,
+    body: JSON.stringify(body),
+  }
+
+  return fetch(url, init).then(handleResponse)
+}
+
+/**
+ * Get a production
+ */
+export const get = async (id: string) => {
+  const getDebug = debug.extend('get')
+  getDebug('id: %s', id)
+
+  const production = await call(`/production/${id}.json`, 'GET').catch(
+    (error) => {
+      if (error.constructor.name === 'AuphonicError') {
```

Comment:
Why do you swallow the error here? Wouldn't it be better to also throw Auphonic errors?